pkg/products/amqonline/reconciler.go

```diff
@@ -232,17 +242,127 @@ func (r *Reconciler) reconcileTemplates(ctx context.Context, serverClient k8scli
-		as.Namespace = r.Config.GetNamespace()
-		owner.AddIntegreatlyOwnerAnnotations(as, r.inst)
-		err := serverClient.Create(ctx, as)
-		if err != nil && !k8serr.IsAlreadyExists(err) {
-			return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("could not create auth service %v: %w", as, err)
+	noneAuthService := &enmasseadminv1beta1.AuthenticationService{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "none-authservice",
+			Namespace: r.Config.GetNamespace(),
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, serverClient, noneAuthService, func() error {
+		owner.AddIntegreatlyOwnerAnnotations(noneAuthService, r.inst)
+		noneAuthService.Spec.Type = "none"
+		return nil
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to create/update 'none' AuthenticationService: %w", err)
+	}
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) reconcileStandardAuthenticationService(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+	r.logger.Info("reconciling standard AuthenticationService")
+
+	const postgresqlName string = "standard-authservice-postgresql"
+
+	// Get CRO to create a Postgresql in the integreatly operator namespace. The
+	// CRO operator SA only has permissions to create the secret in the intly
+	// operator namespace, so it will be first created there, then copied into
+	// the enmasse namespace.
+	_, err := croUtil.ReconcilePostgres(
+		ctx,
+		serverClient,
+		defaultInstallationNamespace,
+		"workshop", // workshop here so that it creates in-cluster postgresql
+		"production",
+		postgresqlName,
+		r.inst.Namespace,
+		postgresqlName,
+		r.inst.Namespace,
+		func(cr metav1.Object) error {
+			owner.AddIntegreatlyOwnerAnnotations(cr, r.inst)
+			return nil
+		},
+	)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("could not create postgresql for standard auth service: %w", err)
+	}
+
+	// Read the CRO secret, to get values to copy to enmasse namespace.
+	croSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.inst.Namespace,
+			Name:      postgresqlName,
+		},
+	}
+	err = serverClient.Get(ctx, k8sclient.ObjectKey{Name: croSecret.Name, Namespace: croSecret.Namespace}, croSecret)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			return integreatlyv1alpha1.PhaseAwaitingComponents, nil
+		}
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	// Copy secret to enmasse namespace (adjust keys for keycloak)
+	keycloakSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.Config.GetNamespace(),
+			Name:      postgresqlName,
+		},
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, serverClient, keycloakSecret, func() error {
+		// I think it would be better to set the owner here to be the
+		// AuthService, but I can't immediately figure out how to get a
+		// reference to the scheme.
+		owner.AddIntegreatlyOwnerAnnotations(keycloakSecret, r.inst)
```

Comment:
@philbrookes @maleck13 half-baked idea: if the RHMI CRD was cluster-scoped, then I think we'd be able to use proper owner-references (even own namespaces?), rather than using annotations and finalizers to managed cleanup ourselves.